derive/src/lib.rs

```diff
@@ -23,7 +32,43 @@ macro_rules! ident {
     };
 }
 
-fn expand_contract(input: LitStr) -> Result<TokenStream> {
+struct ContractArgs {
+    artifact_path: LitStr,
+    runtime_crate: Option<Ident>,
+}
+
+impl Parse for ContractArgs {
+    fn parse(input: ParseStream) -> ParseResult<Self> {
```

Comment:
Is it possible to test these functions maybe even doc tests? This would allow us to easier understand what's happening.

Reply:
Hmm, that is a good question. I can definitely look into it.